extensions/db-diagram/src/models/ITableModel.ts

```diff
@@ -0,0 +1,29 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export default interface ITableModel {
+
+	//Name of the table
+	name: string;
+
+	//Summary of the table
+	summary: string;
+
+	//Map containing columns pointing to their types
+	columns: Map<string, string>;
+
+	//Map containing primary key (or composite primary) pointing to it(s) types
+	primaryKey: Map<string, string>;
+
+	//Map containing foreign key pointing to it(s) types
+	foreignKey: Map<string, string>[];
```

Comment:
Is this the best data structure for this? It's going to be pretty difficult to follow through this in the code.